runtime-common/core/core-types/definition/mixed.cpp

```diff
@@ -1019,6 +1024,55 @@ const string mixed::get_type_str() const {
   return string(get_type_c_str());
 }
 
+std::pair<class_instance<C$ArrayAccess>, bool> try_as_array_access(const mixed &m) noexcept {
```

Comment:
Don't you want to add a new tag `ARRAY_ACCESS_OBJECT` instead?

Reply:
No, I don't.
In this case `ARRAY_ACCESS_OBJECT` would imply `OBJECT` etc, so I think it's overcomplication

Reply:
It makes things more complicated, yep. But it eliminates dynamic cast on each call to a frequently used operation.